### PR TITLE
Feature/scaleio volume movement

### PR DESCRIPTION
**Pull request by thanset**
This feature branch includes JSON parameters to utilize scaleIO mounts underneath the Cassandra FW.

Dvdcli is called right before the executor is started on a node, and mounts an already-provisioned volume underneath it. 

Any rescheduling or node failure will then be resilient to cluster issues.  

Libstorage for rexray must have premption enabled in order to support volume movement properly.

### CODE REVIEW

**Comment by thanset** [comment]
This pr will be squashed before committing.  Some of the commits are debugging on the cluster.

File: cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraTaskExecutor.java [hunk 1]
```diff
@@ -83,49 +78,45 @@ public static final CassandraTaskExecutor parse(
...
-            .setFrameworkId(Protos.FrameworkID.newBuilder()
+            String frameworkId,
+            String name,
+            String role,
+            String principal,
+            ExecutorConfig config) {
+
+        Protos.ExecutorInfo.Builder executorBuilder = Protos.ExecutorInfo.newBuilder();
+
+        /*
```
Comment (thanset) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraTaskExecutor.java]: I need to remediate this to have it work with @aaronjwood's changes in dev.

File: cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/util/TaskUtils.java [hunk 2]
```diff
@@ -31,13 +32,29 @@ public static Environment createEnvironment(
...
         final Set<String> uris) {
-        return uris.stream()
-            .map(uri -> CommandInfo.URI.newBuilder()
-                .setCache(false)
-                .setExecutable(false)
-                .setExtract(true)
-                .setValue(uri).build())
-            .collect(Collectors.toList());
+
+        Collection<CommandInfo.URI> list = new ArrayList<>();
```
Comment (thanset) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/util/TaskUtils.java]: Is there a cleaner way of doing this vs the map() function being done before?

File: cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/CassandraScheduler.java [hunk 3]
```diff
@@ -195,10 +196,22 @@ public void resourceOffers(SchedulerDriver driver,
 
             final Optional<Block> currentBlock = planManager.getCurrentBlock();
 
+            // Filter offers here.
```
Comment (thanset) [inline cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/CassandraScheduler.java]: Can probably do a reduce() and pipe a set of strings into it for "accepted" strings I want to map for. Also need to add params for string I want to filter out by.

File: cassandra-commons/src/test/java/com/mesosphere/dcos/cassandra/common/config/ConfigurationManagerTest.java [hunk 4]
```diff
@@ -158,7 +158,9 @@ public void applyConfigUpdate() throws Exception {
                 17,
                 "/java/home",
                 URI.create("/jre/location"), URI.create("/executor/location"),
-                URI.create("/cassandra/location"));
+                URI.create("/cassandra/location"), URI.create("file:///opt/mesosphere/bin/dvdcli"),
```
Comment (aaronjwood) [inline cassandra-commons/src/test/java/com/mesosphere/dcos/cassandra/common/config/ConfigurationManagerTest.java]: Formatting looks a little off.

File: cassandra-commons/src/test/resources/scheduler.yml [hunk 5]
```diff
@@ -9,6 +9,9 @@ executor:
   jre_location : ${EXECUTOR_JRE_LOCATION:-'https://s3-us-west-2.amazonaws.com/cassandra-framework-dev/jre/linux/server-jre-8u74-linux-x64.tar.gz'}
   executor_location : ${EXECUTOR_LOCATION:-'https://s3-us-west-2.amazonaws.com/cassandra-framework-dev/testing/executor.zip'}
   cassandra_location : ${EXECUTOR_CASSANDRA_LOCATION:-'https://s3-us-west-2.amazonaws.com/cassandra-framework-dev/testing/apache-cassandra-2.2.5-bin.tar.gz'}
+  dvdcli : ${EXECUTOR_DVDCLI:-'file:///opt/mesosphere/bin/dvdcli'}
+  volume_driver : ${EXECUTOR_VOLUME_DRIVER:-"rexray"}
+  volume_name : ${EXECUTOR_VOLUME_NAME:-"timh_cass"}
```
Comment (aaronjwood) [inline cassandra-commons/src/test/resources/scheduler.yml]: What's your idea for having a generic default for this?

File: cassandra-executor/apache-cassandra-2.2.5/conf/cassandra-rackdc.properties [hunk 6]
```diff
@@ -1,4 +1,4 @@
 #DCOS got yo Cassandra!
-#Mon Oct 10 07:37:44 PDT 2016
+#Fri Oct 14 15:19:24 EDT 2016
```
Comment (aaronjwood) [inline cassandra-executor/apache-cassandra-2.2.5/conf/cassandra-rackdc.properties]: Same thing I ran into :)

File: cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/CassandraScheduler.java (hunk 3 above)
Comment (aaronjwood) [inline cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/CassandraScheduler.java]: Will this be conditional and/or more generic (POD1, SDS)?

File: cassandra-scheduler/src/test/java/com/mesosphere/dcos/cassandra/scheduler/TestUtils.java [hunk 7]
```diff
@@ -65,6 +65,8 @@ public static int ipToInt(String ipAddress) {
                 .addResources(ResourceUtils.getUnreservedScalar("cpus", cpu))
                 .addResources(ResourceUtils.getUnreservedScalar("mem", memory))
                 .addResources(ResourceUtils.getUnreservedScalar("disk", disk))
+                .addAttributes(Protos.Attribute.newBuilder().setText(Protos.Value.Text.newBuilder().setValue("SDS")).setName("SDS").setType(Protos.Value.Type.TEXT))
```
Comment (aaronjwood) [inline cassandra-scheduler/src/test/java/com/mesosphere/dcos/cassandra/scheduler/TestUtils.java]: Can this be more generic too so we don't have to always assume ScaleIO?

File: cassandra-commons/src/test/resources/scheduler.yml (hunk 5 above)
Comment (thanset) [inline cassandra-commons/src/test/resources/scheduler.yml]: dvdcli would be snatched from a webserver to reduce the dependency on having it pushed out across the cluster.

I'm not sure about defaults.  We can name it something generic. I need to tie the scaleIO capability to the file system as "mount" vs "root" which should just use the regular disk offers from Mesos.

File: cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/CassandraScheduler.java (hunk 3 above)
Comment (thanset) [inline cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/CassandraScheduler.java]: Yes these will need JSON params as well, those will be added.

File: cassandra-scheduler/src/test/java/com/mesosphere/dcos/cassandra/scheduler/TestUtils.java (hunk 7 above)
Comment (thanset) [inline cassandra-scheduler/src/test/java/com/mesosphere/dcos/cassandra/scheduler/TestUtils.java]: Once I add in the JSON params they'll grab their properties there 👍 .

File: universe/marathon.json.mustache [hunk 8]
```diff
@@ -155,6 +156,9 @@
 ,"EXECUTOR_NETWORK_MODE":"{{executor.network_mode}}"
 ,"EXECUTOR_CNI_NETWORK":"{{executor.cni_network}}"
 ,"EXECUTOR_JAVA_HOME":"./jre"
+,"EXECUTOR_DVDCLI": "{{executor.dvdcli}}",
```
Comment (aaronjwood) [inline universe/marathon.json.mustache]: I just realized these three entries have an extra comma. Will produce broken JSON output.

**Pull request merged**